services/movie_session.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Optional
+from django.db.models import QuerySet
+from db.models import MovieSession, Movie, CinemaHall
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    movie = Movie.objects.get(pk=movie_id)
+    cinema_hall = CinemaHall.objects.get(pk=cinema_hall_id)
+    return MovieSession.objects.create(
+        movie=movie,
+        cinema_hall=cinema_hall,
+        show_time=movie_show_time
+    )
+
+
+def get_movies_sessions(session_date: Optional[str] = None) -> QuerySet:
+    queryset = MovieSession.objects.all()
+    if session_date:
+        queryset = queryset.filter(show_time__date=session_date)
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> Optional[MovieSession]:
+    try:
+        return MovieSession.objects.get(pk=movie_session_id)
+    except MovieSession.DoesNotExist:
+        return None
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: Optional[datetime] = None,
+        movie_id: Optional[int] = None,
+        cinema_hall_id: Optional[int] = None
+) -> Optional[MovieSession]:
+    if not any([show_time, movie_id, cinema_hall_id]):
```

Comment:
Create an empty dict. Then you will check if any of the optional arguments are passed to this function. In case if there are - you will add them to the dict. 
Then after all that - make a condition that if your dict has some values to update you will call update on queryset, and if there are not any you don't do anything